docs/website/01_theory/controlRoutines.md

```diff
@@ -28,3 +28,116 @@ enum class ControlMessages {
+a constraint on the non-linear system of equations. The previously mentioned LoadControl method can also be recreated 
+using this technique. For more details on the standard arc-length method, refer the works 
+of Wempner[@WEMPNER19711581], Crisfield[@CRISFIELD198155], Ramm[@wunderlich_strategies_1981] and 
+Riks[@riks_application_1972] among others. A path following object is constructed as follows:  
+```cpp
+auto alc = Ikarus::PathFollowing(nr, load_steps, stepSize, pft);
+```
+where `#!cpp nr` is a Newton-Raphson solver which considers a scalar subsidiary function and is defined by
+```cpp
+auto nr = Ikarus::makeNewtonRaphsonWithSubsidiaryFunction(nonLinOp, std::move(linSolver));
+```
+and `#!cpp pft` is the desired path following technique. Three different path following techniques are included, namely
+
+* Standard arc-length method
+* Load control method (as a subsidiary function under this generalized implementation)
+* Displacement control method (uses a vector of indices which are all controlled by a same `#!cpp stepSize`).
+
+These can be invoked by defining 
+```cpp
+auto pft = Ikarus::StandardArcLength{psi};
+auto pft = Ikarus::LoadControlWithSubsidiaryFunction{};
+auto pft = Ikarus::DisplacementControl{controlledIndices};
+```
+The default path following type is the `#!cpp Ikarus::StandardArcLength{psi=1.0}`.
+
+It is to note that in the current implementation, it is assumed that the external forces are given by 
+$F_{ext} = F_{ext}^0\lambda$ such that 
+$$
+-\frac{\partial \mathbf{R}}{\partial \lambda} = F_{ext}^0
+$$
+An implementation for a general non-linear $F_{ext} = F_{ext}^0\left(\mathbf{D},\lambda\right)\lambda$ is an open task.
+
+In order to create an own implementation for the scalar subsidiary function, the user has to create a `#!cpp struct` 
+with the following three functions:  
+```cpp
+void evaluateSubsidiaryFunction(SubsidiaryArgs& args);
+void initialPrediction(NonLinearOperator& nonLinearOperator, SubsidiaryArgs& args);
+void intermediatePrediction(NonLinearOperator& nonLinearOperator, SubsidiaryArgs& args);
+```
+The function `#!cpp evaluateSubsidiaryFunction(SubsidiaryArgs& args)` is used to evaluate the subsidiary function and 
+its derivatives with respect to the displacement $\mathbf{D}$ and the load factor $\lambda$ for every Newton-Raphson iteration. The other two functions 
+are used to specify a prediction for $\mathbf{D}$ and $\lambda$ initially for the very first `#!cpp load_steps` and for 
+all the other intermediate subsequent `#!cpp load_steps`.   
+
+`#!cpp SubsidiaryArgs` is a `#!cpp struct` which is defined as
+```cpp
+struct SubsidiaryArgs {
+  double stepSize;
+  Eigen::VectorX<double> DD;
+  double Dlambda{};
+  double f{};
+  Eigen::VectorX<double> dfdDD;
+  double dfdDlambda{};
+};
+```
+An example for the standard arc-length method is shown below:
+```cpp
+struct StandardArcLength {
+  public:
+    explicit StandardArcLength(double p_psi = 1.0) : psi{p_psi} {}
```

Comment:
psi should not be defined in the constructor like this.

Can you add some logic to this like:
as https://godbolt.org/z/x45788j9b .

We can disuss this together